core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -866,24 +864,33 @@ public Response getWebUIWorkers() {
   @GET
   @Path(WEBUI_MASTERS)
   public Response getWebUIMasters() {
-    return RestUtils.call(() -> {
-      MasterWebUIMasters response = new MasterWebUIMasters();
-
-      response.setDebug(Configuration.getBoolean(PropertyKey.DEBUG));
-
-      MasterInfo[] failedMasterInfos = mMetaMaster.getLostMasterInfos();
-      response.setFailedMasterInfos(failedMasterInfos);
-
-      MasterInfo[] normalMasterInfos = mMetaMaster.getMasterInfos();
-      response.setNormalMasterInfos(normalMasterInfos);
-
-      InetSocketAddress leaderMasterAddress = mMasterProcess.getRpcAddress();
-      MasterInfo leaderMasterInfo = new MasterInfo(MASTER_ID_NULL,
-              new Address(leaderMasterAddress.getHostString(), leaderMasterAddress.getPort()),
-              System.currentTimeMillis());
-      response.setLeaderMasterInfo(leaderMasterInfo);
-      return response;
-    }, Configuration.global());
+    Gauge lastCheckpointGauge = MetricsSystem.METRIC_REGISTRY.getGauges()
```

Comment:
this `MetricsSystem.METRIC_REGISTRY.getGauges()` is a bit expensive, can you only get once?

Reply:
Updated. `WEBUI_OVERVIEW` in this class also call `getGauges()` multiple times, but it's not related to this PR.